src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/HoverHandler.cs

```diff
@@ -87,9 +87,8 @@ public async Task<Hover> HandleRequestAsync(TextDocumentPositionParams request,
                 return null;
             }
 
-            var mappingResult = await _documentMappingProvider.MapToDocumentRangeAsync(projectionResult.LanguageKind, request.TextDocument.Uri, result.Range, cancellationToken).ConfigureAwait(false);
-
-            if (mappingResult == null)
+            var mappingResult = await _documentMappingProvider.MapToDocumentRangesAsync(projectionResult.LanguageKind, request.TextDocument.Uri, new[] { result.Range }, cancellationToken).ConfigureAwait(false);
+            if (mappingResult == null || mappingResult.Ranges[0].IsUndefined())
```

Comment:
Would it be possible for `mappingResult != null` but `mappingResult.Ranges.Length == 0` and thus `mappingResult.Ranges[0]` is array out of bounds?

Reply:
That wouldn't happen unless we change the server to behave differently. The current behavior I've defined is that the server will provide a 1:1 mapping for the array passed in. If it can't remap a certain range, it will return an `UndefinedRange` in its place. In fact, this behavior is a requirement if we want the API to accept multiple ranges. Otherwise, we won't know which output range maps to which input range. The other option is for the API to return pairs but that would make things more complicated.